### PR TITLE
chore: Enable Google provider version 5 (remove max version constraint)

### PR DESCRIPTION
**Pull request by MyMirelHub**
## Description

This update includes bumping the Terraform version to 5.12. The primary goal of this change is to ensure compatibility with the latest google api  features and improvements, as well as to maintain up-to-date practices within our infrastructure management.

## Motivation and Context

Major version API upgrades are essential for leveraging the latest functionalities and enhancements offered by the Google provider. This change is driven by the need to stay aligned with the evolving landscape of infrastructure as code (IaC) and to utilize the improved performance and features of the latest google version.

#6 
## How Has This Been Tested?

Followed the contributor's guideline and ran `pre-commit run -a` to ensure the unit tests were passing 

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by michalbil** [comment]
/validate paths="modules/autoscale modules/iam_service_account modules/lb_http_ext_global modules/panorama modules/vpc modules/vpn modules/bootstrap modules/lb_external modules/lb_internal modules/vmseries modules/vpc-peering"

**Comment by michalbil** [comment]
/validate paths="modules/autoscale modules/iam_service_account modules/lb_http_ext_global modules/panorama modules/vpc modules/vpn modules/bootstrap modules/lb_external modules/lb_internal modules/vmseries modules/vpc-peering"

**Comment by michalbil** [comment]
@MyMirelHub there is one issue with `google_secret_manager_secret` argument that needs to be fixed to work with the new provider version, details can be found in validation job logs.

**Comment by pavelrn** [comment]
/validate paths="modules/autoscale modules/iam_service_account modules/lb_http_ext_global modules/panorama modules/vpc modules/vpn modules/bootstrap modules/lb_external modules/lb_internal modules/vmseries modules/vpc-peering"

**Comment by pavelrn** [comment]
/validate paths="modules/autoscale modules/iam_service_account modules/lb_http_ext_global modules/panorama modules/vpc modules/vpn modules/bootstrap modules/lb_external modules/lb_internal modules/vmseries modules/vpc-peering"

**Review by pavelrn — APPROVED** [review]

File: modules/autoscale/versions.tf [hunk 1]
```diff
@@ -2,7 +2,7 @@ terraform {
   required_version = ">= 1.3, < 2.0"
   required_providers {
     google = {
-      version = "~> 4.54"
+      version = ">= 5.0.0"
```
Comment (michalbil) [inline modules/autoscale/versions.tf:5]:
```suggestion
      version = ">= 5"
```

**Review by michalbil — CHANGES_REQUESTED** [review]
@pavelrn can you please update PR title to reflect the changes?

File: modules/autoscale/versions.tf (hunk 1 above)
Comment (pavelrn) [reply modules/autoscale/versions.tf]: Sure!

**Review by pavelrn — APPROVED** [review]

**Review by michalbil — APPROVED** [review]

**Pull request merged**